compiler/rustc_resolve/src/imports.rs

```diff
@@ -693,16 +749,34 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
                 format!("`{path}`")
             })
             .collect::<Vec<_>>();
-        let msg = format!("unresolved import{} {}", pluralize!(paths.len()), paths.join(", "),);
+        let msg = if errors.len() == 1
+            && let Some(message) =
+                errors[0].1.on_unknown_item_attr.as_mut().and_then(|a| a.message.take())
+        {
+            message
+        } else {
+            format!("unresolved import{} {}", pluralize!(paths.len()), paths.join(", "),)
+        };
 
         let mut diag = struct_span_code_err!(self.dcx(), span, E0432, "{msg}");
```

Comment:
I think we should probably refactor this to use structured diagnostics first. Also may be good to check how other diagnostic attrs report their errors if we can make it similar

Reply:
Like... ideally the original error would be a field of the attribute based error so we can handle everything in fluent